resources/tentative/charts/observable-plot.js

```diff
@@ -0,0 +1,265 @@
+        })
+        .sort((stateA, stateB) => stateB.total - stateA.total);
+
+    /* Array<state: string> */
+    const stateSortedArray = stateInformationSortedArray.map(({ state }) => state);
+    /* Array<state: string> */
+    const statesWithMostFlights = stateSortedArray.slice(0, 6);
+
+    // Flatten the information in preparedData.stateInformationSortedArray, so that we
+    // have one item == one airport information.
+    /* Array<{state, iata, name, city, index, origin, destination, total}> */
+    const plotData = stateInformationSortedArray.flatMap(({ mostUsedAirports, total, state }) => {
+        const enrichedMostUsedAirports = mostUsedAirports.map(({ iata, name, city }, index) => ({
+            state,
+            iata,
+            name,
+            city,
+            index, // This will be used to have consistent colors.
+            ...flightsByAirport.get(iata),
+        }));
+        enrichedMostUsedAirports.push({
+            state,
+            iata: "Other",
+            total: total - d3Array.sum(mostUsedAirports, ({ iata }) => flightsByAirport.get(iata)?.total),
+            index: enrichedMostUsedAirports.length,
+        });
+        return enrichedMostUsedAirports;
+    });
+
+    preparedData = {
+        airports,
+        flights,
+        flightsByAirport,
+        byAirport,
+        stateInformationSortedArray,
+        stateSortedArray,
+        statesWithMostFlights,
+        plotData,
+    };
+}
+
+function parseAirportsInformation() {
+    return {
+        airports: csvParse(airportsString),
+        flights: csvParse(flightsString),
+    };
+}
+
+function groupFlightsByAirports(flights) {
+    const flightsByAirport = new Map();
+
+    for (const { origin, destination, count } of flights) {
+        const infoForOriginAirport = flightsByAirport.get(origin) ?? { origin: 0, destination: 0, total: 0 };
+        const intCount = Number(count);
+        if (Number.isNaN(intCount)) {
+            console.error(`Couldn't convert ${count} to number.`);
+            continue;
+        }
+        infoForOriginAirport.origin += intCount;
+        infoForOriginAirport.total += intCount;
+        flightsByAirport.set(origin, infoForOriginAirport);
+        const infoForDestinationAirport = flightsByAirport.get(destination) ?? { origin: 0, destination: 0, total: 0 };
+        infoForDestinationAirport.destination += intCount;
+        infoForDestinationAirport.total += intCount;
+        flightsByAirport.set(destination, infoForDestinationAirport);
+    }
+
+    return flightsByAirport;
+}
+
+function isReady() {
+    return !!preparedData;
+}
+
+function addStackedBars() {
+    if (!isReady())
+        throw new Error("Please prepare the data first.");
+
+    const options = {
+        ...DEFAULT_OPTIONS,
+        color: { type: "categorical" },
+        x: {
+            domain: preparedData.stateSortedArray,
+        },
+        y: { grid: true, tickFormat: "~s" },
+        marks: [
+            // stacked bars
```

Comment:
This comment seems unnecessary given this function is called addStackedBars.

Reply:
This specifically means that the next line is about the bars themselves, as opposed to the following lines.